TypeScript/app/gilded-rose.ts

```diff
@@ -10,6 +12,127 @@ export class Item {
   }
 }
 
+interface ItemUpdateStrategy {
```

Comment:
It is possible (and preferable) to move all the strategy related code into separate file(s). However, it is not practical (still possible) with the `Item` class defined inside this file. Since we can't touch Item, I decided to keep everything in a single file. If we could extract `Item` into its own file then I'd split the rest.